simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1301 @@
+
+        Returns
+        -------
+        dict:
+            A dictionary with the astropy units.
+        """
+
+        # Build a dictionary with astropy units for the unit of the event's (header's) parameters.
+        all_event_astropy_units = {}
+        for key in parameters[1:]:  # starting at the second
+            # element to avoid the non-numeric (e.g. 'EVTH') key.
+
+            # We extract the astropy unit (dimensionless in case no unit is provided).
+            if key in non_astropy_units:
+                with cds.enable():
+                    unit = u.Unit(non_astropy_units[key])
+            else:
+                unit = u.dimensionless_unscaled
+            all_event_astropy_units[key] = unit
+        return all_event_astropy_units
+
+    @property
+    def telescope_indices(self):
+        """
+        The telescope index (or indices), which are considered for the production of the histograms.
+
+        Returns
+        -------
+        list:
+            The indices of the telescopes of interest.
+        """
+        return self._telescope_indices
+
+    @telescope_indices.setter
+    def telescope_indices(self, telescope_new_indices):
+        """
+        Set the telescope index (or indices).
+        If self.individual_telescopes is True, the indices of the telescopes passed are analyzed
+        individually (different histograms for each telescope) even if all telescopes are listed.
+
+        Parameters
+        ----------
+        telescope_new_indices: int or list of int or np.array of int
+            The indices of the specific telescopes to be inspected. If not specified, all telescopes
+            are treated together in one histogram and the value of self._telescope_indices is a list
+            of all telescope indices.
+
+        Raises
+        ------
+        TypeError:
+            if the indices passed through telescope_index are not of type int.
+        """
+
+        if telescope_new_indices is None:
+            self._telescope_indices = np.arange(self.num_telescopes)
+        else:
+            if not isinstance(telescope_new_indices, (list, np.ndarray)):
+                telescope_new_indices = np.array([telescope_new_indices])
+            for i_telescope in telescope_new_indices:
+                if not isinstance(i_telescope, (int, np.integer)):
+                    msg = "The index or indices given are not of type int."
+                    self._logger.error(msg)
+                    raise TypeError
+            self._telescope_indices = np.sort(telescope_new_indices)
+
+    @property
+    def hist_config(self):
+        """
+        The configuration of the histograms.
+
+        Returns
+        -------
+        dict:
+            the dictionary with the histogram configuration.
+        """
+        if self._hist_config is None:
+            msg = "No configuration was defined before. The default config is being created now."
```

Comment:
Would it make sense to say `No histogram configuration...`, as there are other configuration steps in the program and this might be confused.